src/Menu/PageUrlProvider.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of Monsieur Biz' Cms Page plugin for Sylius.
+ *
+ * (c) Monsieur Biz <sylius@monsieurbiz.com>
+ *
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace MonsieurBiz\SyliusCmsPagePlugin\Menu;
+
+use MonsieurBiz\SyliusCmsPagePlugin\Entity\PageInterface;
+use MonsieurBiz\SyliusCmsPagePlugin\Repository\PageRepositoryInterface;
+use MonsieurBiz\SyliusMenuPlugin\Provider\AbstractUrlProvider;
+use Symfony\Component\Routing\RouterInterface;
+use Webmozart\Assert\Assert;
+
+class PageUrlProvider extends AbstractUrlProvider
+{
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** for this case, I've already seen (in Symfony?) a workaround with a "middle" class:

```php
if (class_exists(AbstractUrlProvider::class)) {
	class MiddlePageUrlProvider extends AbstractUrlProvider
	{}
} else {
	class MiddlePageUrlProvider
	{}
}

class PageUrlProvider extends MiddlePageUrlProvider
{
```

Reply:
![image](https://github.com/monsieurbiz/SyliusCmsPagePlugin/assets/11380627/39338563-d9b2-44cd-9e1a-cfba5ef30561)

Reply:
yes!